core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/mapper/AccountMapper.kt

```diff
@@ -0,0 +1,31 @@
+package org.mifos.mobilewallet.core.data.fineract.entity.mapper
+
+import com.mifos.mobilewallet.model.entity.accounts.savings.SavingAccount
+import com.mifos.mobilewallet.model.entity.client.ClientAccounts
+import com.mifos.mobilewallet.model.domain.Account
+import javax.inject.Inject
+
+class AccountMapper @Inject constructor(private val currencyMapper: CurrencyMapper) {
+
+    fun transform(clientAccounts: ClientAccounts?): List<Account> {
+        val accountList = mutableListOf<Account>()
+
+        if (clientAccounts != null && clientAccounts.savingsAccounts != null
+            && clientAccounts.savingsAccounts.isNotEmpty()) {
+
+            for (savingAccount in clientAccounts.savingsAccounts) {
+                val account = Account().apply {
```

Comment:
You are setting all the values, you don't need to use `apply` here directly use primary contractor.

Reply:
no image field is missing , so i prefer using this one ..